src/app/challenges/[slug]/ChallengeViewer.tsx

```diff
@@ -1,26 +1,24 @@
-"use client";
-
+import { MDXComponents } from "mdx/types";
+import { MDXRemote } from "next-mdx-remote/rsc";
 import MDXCodeHighlighter from "@/components/MDXCodeHighlighter";
 import MDXHeaders from "@/components/MDXHeaders";
-
-import MarkdownViewer from "@/components/MarkdownViewer";
-import CrossDetection from "@/components/CrossDetection";
+import { readPublicDataSync } from "@/utils/fs";
 
 const ChallengeMDXCodeHighlighter = MDXCodeHighlighter();
 
-const ChallengeViewer = (props) => {
-  const { data } = props;
+const ChallengeViewer = ({ params }) => {
+  const data = readPublicDataSync(`challenges/${params.slug}.mdx`);
 
   return (
-    <CrossDetection
-      dataSource={data}
-      className="challenge-viewer py-[30px] sm:py-[40px] md:py-[60px]"
-    >
-      <MarkdownViewer
-        data={data}
-        components={{ ...ChallengeMDXCodeHighlighter, ...MDXHeaders }}
-      ></MarkdownViewer>
-    </CrossDetection>
+    <MDXRemote
```

Comment:
use `Suspense` to wrap the time-consuming server components rather than adding a loading page, as well as other viewers